matrix/src/dense.rs

```diff
@@ -162,8 +162,8 @@ impl<'a, T> RowMajorMatrixViewMut<'a, T> {
         &mut self.values[r * self.width..(r + 1) * self.width]
     }
 
-    pub fn rows(&self) -> impl Iterator<Item = &[T]> {
-        self.values.chunks_exact(self.width)
+    pub fn rows(&mut self) -> impl Iterator<Item = &mut [T]> {
```

Comment:
How about a separate `rows_mut()`? Seems idiomatic to have both, in case some caller wants to do multiple immutable borrows